src/Compatibility/Microsoft.DotNet.PackageValidation/ReadMe.md

```diff
@@ -0,0 +1,198 @@
+
+# How To Add Package Validation To Your Projects
+
+Package Validation is currently being shipped as a msbuild sdk package which can be consumed in the csproj. It is set of tasks and targets that run after the dotnet pack
+The csproj will look like
+
+```C#
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <Sdk Name="Microsoft.DotNet.PackageValidation" Version="1.0.0-preview4.21121.3" />
+</Project>
+```
+
+## Scenarios and User Experience
+
+The first preview version of the package validation covers the basic scenarios for package validation. They are as follows:-
+
+## Validation Among Compatible Frameworks
+
+Package containing compatible frameworks need to ensure that code compiled against one can run against other. Examples of compatible pairs will be .NET Standard2.0 and .NET 6.0, .NET 5.0 and .NET 6.0 . In both of these cases, the users can compile against .NET Standard 2.0 or NET 5.0 and try to run on .NET 6.0, if the binaries are not compatible the users might end up with runtime errors.
+
+Package Validation will catch these errors at pack time. eg.
+
+Finley is working on a game which do a lot of string manipulation. They need to support both .NET Framework and .NET Core customers. They started with just targeting .NET Standard 2.0 but now they realize they want to take advantage of the spans in .NET Core 3.0 and above to avoid unnecessary string allocations. In order to do that, they are multi-targeting for NET Standard2.0 and .NET 6.0,.
+
+Finley has written the following code:
+```c#
+#if NET6_0_OR_GREATER
+    public void DoStringManipulation(ReadOnlySpan<char> input)
+    {
+        // use spans to do string operations.
+    }
+#else
+    public void DoStringManipulation(string input)
+    {
+        // Do some string operations.
+    }
+#endif
+```
+
+When Finley packs the project it fails with the following error:
+
+```c#
+error : .NETStandard,Version=v2.0 assembly api surface area should be compatible with net6.0 assembly surface area so we can compile against .NETStandard,Version=v2.0 and run on net6.0 .framework.
+error : API Compatibility errors between lib/netstandard2.0/A.dll (left) and lib/net6.0/A.dll (right):
+error CP0002: Member 'A.B.DoStringManipulation(string)' exists on the left but not on the right.
+```
+
+Finley understands that they shouldn't exclude ```DoStringManipulation(string)``` but instead just provide an additional ```DoStringManipulation(ReadOnlySpan<char>)``` method for .NET 6.0 and changes the code accordingly:
+
+```c#
+#if NET6_0_OR_GREATER
+    public void DoStringManipulation(ReadOnlySpan<char> input)
+    {
+        // use spans to do string operations.
+    }
+#endif
+    public void DoStringManipulation(string input)
+    {
+        // Do some string operations.
+    }
+```
```

Comment:
Perhaps in here include another example after the fix of how calling dotnet pack now shows a successful build.